### PR TITLE
Adding squid pack support for OneOps

### PR DESCRIPTION
**Pull request by msumankumar**

### CODE REVIEW

**Comment by ranand19** [comment]
acl handling is missing from recipe. need to add the acl handler libraries and use it.

File: components/cookbooks/squid/doc/index.md [hunk 1]
```diff
@@ -0,0 +1 @@
+tbd
```
Comment (gauravlall) [inline components/cookbooks/squid/doc/index.md]: Please add documentation

File: packs/squid.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+:monitors => {
+         'URL' => {:description => 'URL',
+                   :source => '',
+                   :chart => {'min' => 0, 'unit' => ''},
+                   :cmd => 'check_http_status!#{cmd_options[:host]}!#{cmd_options[:port]}!#{cmd_options[:url]}!#{cmd_options[:wait]}!#{cmd_options[:expect]}!#{cmd_options[:regex]}',
+                   :cmd_line => '/opt/nagios/libexec/check_http_status.sh $ARG1$ $ARG2$ "$ARG3$" $ARG4$ "$ARG5$" "$ARG6$"',
+                   :cmd_options => {
+                       'host' => 'localhost',
+                       'port' => '80',
+                       'url' => '/nexus/index.html',
```
Comment (gauravlall) [inline packs/squid.rb]: we should remove  /nexus hardcoding from  here

File: components/cookbooks/squid/metadata.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+attribute 'maximum_object_size',
+  :description => "Maximum Object Size",
+  :default => "512",
+  :format => {
+    :help => 'Maximum file size in MB',
+    :category => '4.Caching',
+    :order => 4
+  }
+
+recipe "start", "Start Squid"
```
Comment (gauravlall) [inline components/cookbooks/squid/metadata.rb]: Add status recipe to support status

File: components/cookbooks/squid/metadata.rb [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+version          "0.1"
+maintainer       "OneOps"
+maintainer_email "support@oneops.com"
+license          "Copyright OneOps, All rights reserved."
+
+grouping 'default',
+  :access => "global",
+  :packages => [ 'base', 'mgmt.catalog', 'mgmt.manifest', 'catalog', 'manifest', 'bom' ]
+
+attribute 'install_type',
```
Comment (gauravlall) [inline components/cookbooks/squid/metadata.rb]: Need to add version

File: components/cookbooks/squid/templates/default/squid.conf.erb [hunk 5]
```diff
@@ -0,0 +1,107 @@
+# This file generated from a Chef template.
+# squid/templates/default/squid.conf.erb
+
+<%= "acl all src" if node['squid']['version'].to_i <3 %>
+
+<% if node['squid']['accel'] == 'true' %>
```
Comment (gauravlall) [inline components/cookbooks/squid/templates/default/squid.conf.erb]: I dont see this 'accel' being set , could you please check ?

**Pull request merged**